tests/core/operations/test_spatialdata_operations.py

```diff
@@ -282,17 +306,15 @@ def test_get_item(points):
         _ = points["not_present"]
 
 
-def test_set_item(full_sdata):
+def test_set_item(full_sdata: SpatialData) -> None:
     for name in ["image2d", "labels2d", "points_0", "circles", "poly"]:
         full_sdata[name + "_again"] = full_sdata[name]
-        with pytest.raises(KeyError):
+        with pytest.warns(UserWarning):
+            full_sdata[name] = full_sdata[name]
+    for name in ["image2d", "labels2d", "points_0"]:
+        # trying to overwrite the file used for backing (only for images, labels and points)
+        with pytest.warns(UserWarning):
             full_sdata[name] = full_sdata[name]
```

Comment:
to be removed since we already test this above and we don't risk overwriteing lazy data now with the __setitem__ call.